frontend/src/pages/shipments/ShipmentForm.tsx

```diff
@@ -0,0 +1,169 @@
+import { gql, useQuery } from '@apollo/client'
+import React, { FunctionComponent, ReactNode, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import Button from '../../components/Button'
+import SelectField, { SelectOption } from '../../components/forms/SelectField'
+import TextField from '../../components/forms/TextField'
+import { MONTH_OPTIONS } from '../../data/constants'
+import {
+  Group,
+  GroupType,
+  Shipment,
+  ShipmentStatus,
+  ShipmentUpdateInput,
+  ShippingRoute,
+} from '../../types/api-types'
+
+interface Props {
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The values to display in the fields of the form. Note that this is NOT a
+   * controlled component.
+   */
+  defaultValues?: Shipment
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: ShipmentUpdateInput) => void
+}
+
+const STATUS_OPTIONS = [
+  { label: 'Abandoned', value: ShipmentStatus.Abandoned },
+  { label: 'Announced', value: ShipmentStatus.Announced },
+  { label: 'Complete', value: ShipmentStatus.Complete },
+  { label: 'In progress', value: ShipmentStatus.InProgress },
+  { label: 'Open', value: ShipmentStatus.Open },
+  { label: 'In staging', value: ShipmentStatus.Staging },
+]
+
+const SHIPPING_ROUTE_OPTIONS = Object.keys(ShippingRoute).map((routeKey) => ({
+  label: (ShippingRoute as any)[routeKey],
+  value: (ShippingRoute as any)[routeKey],
+}))
+
+const GET_ALL_GROUPS = gql`
+  query Groups {
+    listGroups {
+      id
+      name
+      groupType
+    }
```

Comment:
Finally starting to see the upside of GraphQL: only fetching the fields we need, and nothing else.

Reply:
Yeah I'm still waiting to see the benefits, it's still pretty unfamiliar. I guess the main positive thing seems to be standardized type definitions for everything which I've never tried to do with a REST API. I'm too used to protocol buffers at this point.